tests/sentry/api/endpoints/test_project_rule_details.py

```diff
@@ -689,65 +689,66 @@ def test_rule_form_not_valid(self):
-
-            url = reverse(
-                "sentry-api-0-project-rule-details",
-                kwargs={
-                    "organization_slug": project.organization.slug,
-                    "project_slug": project.slug,
-                    "rule_id": rule.id,
-                },
-            )
-            response = self.client.put(
-                url,
-                data={
-                    "name": "hello world",
-                    "actionMatch": "any",
-                    "filterMatch": "any",
-                    "conditions": [],
-                    "actions": [{"id": "sentry.rules.actions.notify_event.NotifyEventAction"}],
-                },
-                format="json",
-            )
-
-            assert response.status_code == 400, response.content
-
-        def test_rule_form_missing_action(self):
-            self.login_as(user=self.user)
-
-            project = self.create_project()
-
-            rule = Rule.objects.create(project=project, label="foo")
-
-            url = reverse(
-                "sentry-api-0-project-rule-details",
-                kwargs={
-                    "organization_slug": project.organization.slug,
-                    "project_slug": project.slug,
-                    "rule_id": rule.id,
-                },
-            )
-            response = self.client.put(
-                url,
-                data={
-                    "name": "hello world",
-                    "actionMatch": "any",
-                    "filterMatch": "any",
-                    "action": [],
-                    "conditions": [
-                        {"id": "sentry.rules.conditions.tagged_event.TaggedEventCondition"}
-                    ],
-                },
-                format="json",
-            )
-
-            assert response.status_code == 400, response.content
+    def test_rule_form_owner_perms(self):
+        self.login_as(user=self.user)
+
+        project = self.create_project()
+
+        rule = Rule.objects.create(project=project, label="foo")
+
+        url = reverse(
+            "sentry-api-0-project-rule-details",
+            kwargs={
+                "organization_slug": project.organization.slug,
+                "project_slug": project.slug,
+                "rule_id": rule.id,
+            },
+        )
+        other_user = self.create_user()
+        response = self.client.put(
+            url,
+            data={
+                "name": "hello world",
+                "actionMatch": "any",
+                "filterMatch": "any",
+                "conditions": [{"id": "sentry.rules.conditions.tagged_event.TaggedEventCondition"}],
+                "actions": [],
+                "owner": other_user.actor.get_actor_identifier(),
+            },
+            format="json",
+        )
+
+        assert response.status_code == 400, response.content
+        assert str(response.data["owner"][0]) == "User is not a member of this organization"
```

Comment:
@wedamija shouldn't we give a 404 or something as to not give away that this organization exists if the user isn't part of it?

Reply:
The organization will already be validated at this point, so that's not a concern since the user will only hit this point if they're a member of the org. It might give away that a given user id or team id is a valid id, but it's easy enough to create a new user or team and assume that every id between 0 -> <new_id> is likely used, so we're not giving much away there either.

Reply:
@wedamija I'm confused. If the user is not a member of the organization and they hit this endpoint, are they going to get a 400 or a 404?

Reply:
oh duh...I understand now. The owner is necessarily the user making the request.

Reply:
This pr isn't changing that. There are two users involved here:
 - The user making the request. This user has to be authenticated and authorized to access the organization and project to create/modify alert rules on the project. I'm not changing this validation at all.
  - The owner of the rule. This is a parameter that the user can provide, and can be a team or a user. Previously this could be any sentry user/team, and now it's restricted to only users and teams that belong to the org. If the passed owner is from another org, then we will return a 400 here.